jaxopt/_src/tree_util.py

```diff
@@ -58,6 +58,10 @@ def tree_vdot(tree_x, tree_y):
   return tree_reduce(operator.add, vdots)
 
 
+def tree_dot(tree_x, tree_y):
```

Comment:
Add docstring (what does it mean to multiply two pytrees?)

Reply:
I added:

_Compute leaves-wise dot product between pytree of arrays._
  
  _Useful to store block diagonal linear operator: each leaf of the tree
  corresponds to a block._